docs/source/services/timelock_service/reverse-migration.rst

```diff
@@ -0,0 +1,133 @@
+embedded service.
+
+The reverse migration process must be run offline (with no AtlasDB clients for the client you are reverse-migrating
+running). We do not currently protect against accidentally not running this process offline.
+Throughout this document, we assume the AtlasDB client you are trying to reverse-migrate is ``client`` .
+
+#. Shut down your AtlasDB clients for ``client`` .
+#. Take a restore timestamp.
+#. Update your key-value service to the restore timestamp.
+#. Revert your AtlasDB client configuration.
+#. Remove your client from the list of TimeLock clients.
+#. Restart the TimeLock servers.
+#. Start your AtlasDB clients.
+
+Step 1: Shut Down AtlasDB Clients
+---------------------------------
+
+Make sure that your AtlasDB clients for ``client`` are shut down. Failure to do this can result in severe data
+corruption, because we cannot guarantee that the Timelock Server and the embedded AtlasDB timestamp services will
+together issue monotonically increasing timestamps.
+
+Step 2: Take a Restore Timestamp
+--------------------------------
+
+Obtain a fresh timestamp for ``client`` from the TimeLock servers. This is important to ensure that the sequence
+of timestamps across TimeLock and the key-value service backed services is monotonically increasing.
+
+This can be obtained using ``curl`` as follows. Note that if you obtain an HTTP 503 error, this is because you curled a
+node that was not the leader; please try the other nodes.
+
+.. code-block:: none
+
+   $ curl -XPOST localhost:8421/client/timestamp/fresh-timestamp
+   123456789
+
+Note down the value returned from this call; we'll refer to it as ``TS`` from here on.
+
+Step 3: Update your Key-Value Service
+-------------------------------------
+
+Update the timestamp table in your key value service, such that it is valid and has the bound set to ``TS`` .
+This will vary depending on your choice of key-value service:
+
+- If you are using **Cassandra**, then you can carry out the following steps.
+  We suppose that your clients have the keyspace ``client`` in the examples below; at a high level, we want to truncate
+  the timestamp table to remove all entries, and then add a single entry with row and column names ``0x7473``
+  (case-sensitive) and a blob value indicating the actual timestamp.
+
+  .. code-block:: none
+
+     cqlsh> USE client;
+     cqlsh:client> SELECT * FROM "_timestamp" ;
+
+      key    | column1      | column2 | value
+     --------+--------------+---------+--------------------
+      0x7473 | 0x6f6c645473 |      -1 | 0x0000000006d30af4
+      0x7473 |       0x7473 |      -1 |               0x00
+
+     (2 rows)
+
+     cqlsh:client> TRUNCATE "_timestamp";
+     cqlsh:client> SELECT * FROM "_timestamp" ;
+
+      key | column1 | column2 | value
+     -----+---------+---------+-------
+
+     (0 rows)
+
+  You will need to encode ``TS`` into an 8-byte hex representation; this can be done using the ``printf`` command:
```

Comment:
Agree that users might put this into clqsh, although the `$` is going to help. Can we say "using the `printf` shell command"?